src/pages/home/ReportScreen.tsx

```diff
@@ -275,7 +275,8 @@ function ReportScreen({
         Performance.markStart(CONST.TIMING.CHAT_RENDER);
     }
     const [isComposerFocus, setIsComposerFocus] = useState(false);
-    const viewportOffsetTop = useViewportOffsetTop(Browser.isMobileSafari() && isComposerFocus && !modal?.willAlertModalBecomeVisible);
+    const shouldAdjustScrollView = useMemo(() => Browser.isMobileSafari() && isComposerFocus && !modal?.willAlertModalBecomeVisible, [isComposerFocus, modal]);
```

Comment:
`index.ts` is only a web file (since there is also an `index.native.ts`), which means that the `Browser.isMobileSafari()` shouldn't really be here. I think this browser check should be moved inside of `index.ts` so that the logic is better incapsulated. The browser check should be coupled with an early return.